packages/noir-ethereum-api/src/util/object.test.ts

```diff
@@ -15,3 +15,10 @@ describe('updateNestedField', () => {
     expect((object as any).x[0]?.y?.z).to.eq(5);
   });
 });
+
+describe('copy', () => {
+  it('copy object', () => {
+    const obj = { a: 1, b: [{ c: 'x' }] };
+    expect(copy(obj)).toStrictEqual(obj);
```

Comment:
We should also check that if you modify the initial object - we don't get the same modifications in a copied one. Otherwise copy could be equal to id function

Reply:
added test